app/decorators/petition_search.rb

```diff
@@ -0,0 +1,75 @@
+class PetitionSearch
+
+  delegate :previous_page, :next_page, :to => :petitions
+  delegate :current_page, :per_page, :to => :petition_results
+  delegate :total_entries, :total_pages, :to => :petition_results
+  delegate :each, :empty?, :offset, :size, :to => :petition_results
+  delegate :first, :length, :to_a, :to_ary, :to => :petition_results
+
+  def initialize(params)
+    @params = params
+  end
+
+  def state
+    State::SEARCHABLE_STATES.include?(@params[:state]) ? @params[:state] : 'open'
+  end
+
+  def search_term
+   @params[:q] || ''
+  end
+
+  def result_count_for_state(state)
+    @petition_search_counts ||= execute_result_counts_query
+    @facets ||= @petition_search_counts.facet(:state).rows
+    default = -> { OpenStruct.new(count: 0) }
```

Comment:
OpenStruct is great for prototyping, but shouldn't leak out into production code.  The implementation you have is nice (keen to avoid ``try`` if we can) - we just need a different thing to work as the default.  Could use an empty array, because that has ``count`` that returns ``0``, but it's a bit opaque.  I'd probably create an explicit ``NullFacet`` class that has the right API and returns ``0`` or empty string in all the right places.